### PR TITLE
refactor(lib): Remove experimental Next implementation

### PR DESCRIPTION
**Pull request by MrRefactor**
# Summary
Remove experimental `scrollview` based `react-native-pager-view` implementation.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)

### CODE REVIEW

**Review by krozniata — APPROVED** [review]

File: ios/RNCPagerView.h [hunk 1]
```diff
@@ -2,22 +2,29 @@
...
 #import <React/UIView+React.h>
 #import <UIKit/UIKit.h>
+#import "UIView+isHorizontalRtlLayout.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface RNCPagerView: UIView
+@interface RNCPagerView: UIView <RtlLayoutProtocol>
 
-- (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol> )eventDispatcher;
```
Comment (okwasniewski) [inline ios/RNCPagerView.h:10]: Is this change required? I think having the protocol here is better

File: ios/RNCPagerView.h [hunk 2]
```diff
@@ -2,22 +2,29 @@
...
+- (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher;
 
 @property(nonatomic) NSInteger initialPage;
-@property(nonatomic) NSString* orientation;
+@property(nonatomic) NSInteger lastReportedIndex;
+@property(nonatomic) NSInteger destinationIndex;
+@property(nonatomic) NSInteger currentIndex;
+@property(nonatomic) NSInteger pageMargin;
+@property(nonatomic, readonly) BOOL scrollEnabled;
```
Comment (okwasniewski) [inline ios/RNCPagerView.h:18]: Why is this now exposed as public API? This should be hidden in @implementation as before

**Review by troZee — APPROVED** [review]

File: ios/Fabric/RNCPagerViewComponentView.h [hunk 3]
```diff
@@ -4,10 +4,19 @@
...
-@interface RNCPagerViewComponentView : RCTViewComponentView <UIScrollViewDelegate>
+@interface RNCPagerViewComponentView : RCTViewComponentView <UIPageViewControllerDataSource, UIPageViewControllerDelegate, UIScrollViewDelegate, RtlLayoutProtocol>
+
+@property(strong, nonatomic, readonly) UIPageViewController *nativePageViewController;
+@property(nonatomic, strong) NSMutableArray<UIViewController *> *nativeChildrenViewControllers;
+@property(nonatomic) NSInteger initialPage;
+@property(nonatomic) NSInteger currentIndex;
+@property(nonatomic) NSInteger destinationIndex;
+@property(nonatomic) NSString* layoutDirection;
+@property(nonatomic) BOOL overdrag;
```
Comment (okwasniewski) [inline ios/Fabric/RNCPagerViewComponentView.h:19]: Same here, all of this shouldn't be public.

**Review by okwasniewski — CHANGES_REQUESTED** [review]

File: ios/RNCPagerViewManager.m [hunk 4]
```diff
@@ -8,12 +8,14 @@ @implementation RNCPagerViewManager
 RCT_EXPORT_MODULE(RNCViewPager)
 
 RCT_EXPORT_VIEW_PROPERTY(initialPage, NSInteger)
-RCT_EXPORT_VIEW_PROPERTY(orientation, NSString)
+RCT_EXPORT_VIEW_PROPERTY(pageMargin, NSInteger)
 
+RCT_EXPORT_VIEW_PROPERTY(orientation, UIPageViewControllerNavigationOrientation)
```
Comment (okwasniewski) [inline ios/RNCPagerViewManager.m:13]: Will this work this way? We probably need RCTConvert for this, what determines the orientation conversion to `UIPageViewControllerNavigationOrientation`

File: ios/RNCPagerViewManager.m [hunk 5]
```diff
@@ -61,7 +63,7 @@ - (void) changeScrollEnabled
     [self goToPage:reactTag index:index animated:false];
 }
 
-RCT_EXPORT_METHOD(setScrollEnabled
+RCT_EXPORT_METHOD(setScrollEnabledImperatively
```
Comment (okwasniewski) [inline ios/RNCPagerViewManager.m:66]: Why do we re-name this method?

File: ios/RNCPagerView.h (hunk 2 above)
Comment (MrRefactor) [reply ios/RNCPagerView.h:18]: to be addressed in followup PR

File: ios/RNCPagerViewManager.m (hunk 4 above)
Comment (MrRefactor) [reply ios/RNCPagerViewManager.m:13]: to be addressed in followup PR

**Review by okwasniewski — APPROVED** [review]

**Review by troZee — APPROVED** [review]

**Comment by OlegWock** [comment]
Hello! Would you mind me asking why you decided to not proceed with this implementation?

I'm using react-native-tab-view and currently facing [this issue](https://github.com/callstack/react-native-pager-view/issues/575) in my app on iOS, after some debugging I figured it's coming from react-native-pager-view. I tried a lot of stuff, but in the end what worked for me was to switch to new implementation (one removed in this PR) by monkey-patching file inside node_modules. So I was wondering if there is any (potential?) problems with new next implementation?

**Comment by sriharsha-y** [comment]
@MrRefactor Can I know what's the potential problems with this new `scrollview` based implementation and why it is removed ?

**Comment by MrRefactor** [comment]
> Hello! Would you mind me asking why you decided to not proceed with this implementation?
> 
> I'm using react-native-tab-view and currently facing [this issue](https://github.com/callstack/react-native-pager-view/issues/575) in my app on iOS, after some debugging I figured it's coming from react-native-pager-view. I tried a lot of stuff, but in the end what worked for me was to switch to new implementation (one removed in this PR) by monkey-patching file inside node_modules. So I was wondering if there is any (potential?) problems with new next implementation?

Hey, please open an issue for it in this repo
